src/Admin/TeamAdmin.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace App\Admin;
+
+use App\Entity\Administrator;
+use App\Entity\Team\Team;
+use App\Form\Admin\MemberAdherentAutocompleteType;
+use App\Form\Admin\MemberType;
+use App\Team\TeamMemberHistoryManager;
+use App\Team\TypeEnum;
+use Sonata\AdminBundle\Admin\AbstractAdmin;
+use Sonata\AdminBundle\Datagrid\DatagridMapper;
+use Sonata\AdminBundle\Datagrid\ListMapper;
+use Sonata\AdminBundle\Form\FormMapper;
+use Sonata\DoctrineORMAdminBundle\Filter\ChoiceFilter;
+use Sonata\DoctrineORMAdminBundle\Filter\ModelAutocompleteFilter;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\CollectionType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Security\Core\Security;
+
+class TeamAdmin extends AbstractAdmin
+{
+    private $security;
+    private $teamMemberHistoryManager;
+
+    /** @var Team|null */
+    private $beforeUpdate;
+
+    public function __construct(
+        $code,
+        $class,
+        $baseControllerName = null,
+        Security $security,
+        TeamMemberHistoryManager $teamMemberHistoryManager
+    ) {
+        parent::__construct($code, $class, $baseControllerName);
+
+        $this->security = $security;
+        $this->teamMemberHistoryManager = $teamMemberHistoryManager;
+    }
+
+    protected function configureFormFields(FormMapper $formMapper)
+    {
+        $formMapper
+            ->with('Informations ⚙️', ['class' => 'col-md-6'])
+                ->add('type', ChoiceType::class, [
+                    'label' => 'Type',
+                    'choices' => TypeEnum::ALL,
+                    'choice_label' => function (string $choice) {
+                        return "team.type.$choice";
+                    },
```

Comment:
we can create a FormType for `TeamTypeChoice` and re use it in `configureDatagridFilters`?